backend/utills/http.utills.ts

```diff
@@ -0,0 +1,34 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { allPublishedProfilesPayload } from '../profile/profile.service'
+
+export async function allRequestController(
+  req: NextApiRequest,
+  callbackFn: () => Promise<void>,
+) {
+  switch (req.method) {
+    case 'GET':
+      return callbackFn()
+    case 'POST':
+      return callbackFn()
+    default:
+      throw new Error('Unsupported HTTP method')
+  }
+}
+
+export async function getRequestController(
+  req: NextApiRequest,
+  res: NextApiResponse,
+) {
+  try {
```

Comment:
**issue:** Utilities should not know anything about any domain. Getting `allPublishedProfilesPayload` is not a part of utility, but rather `/profiles/route.ts`. Move it, and adjust it there please after you refactor this `allRequestController`